rudder-web/src/main/scala/com/normation/rudder/web/rest/RestExtractorService.scala

```diff
@@ -679,5 +713,27 @@ case class RestExtractorService (
     }
   }
 
+  def extractQuery (params : Map[String,List[String]]) : Box[Option[Query]] = {
+    extractOneValue(params,"query")(convertToQuery) match {
+      case Full(None) =>
+        extractOneValue(params,CRITERIA)(convertToQueryCriterion) match {
+          case Full(None) => Full(None)
+          case Full(Some(criterion)) =>
+            for {
+              optType <- extractOneValue(params,TARGET)(convertToQueryReturnType)
```

Comment:
We get only a Full(None) only if target is not set (and so we set the default value below)

If target is not correctly parsed we get a Failure, so we wont have a problem

Semantic of the extract function are:
Full(Some(x)) => The param was present and correct
Full(None) => The param was missing and should be treated (keep as None, use a default value, error for mandatory params)
Failure/Empty => the param was present but could not correctly been parsed